transport/http/handler.go

```diff
@@ -93,8 +96,20 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request) error {
 		ctx = baggage.NewContextWithHeaders(ctx, headers.Items())
 	}
 
+	start := time.Now()
+
 	// TODO capture and handle panic
-	return h.Handler.Handle(ctx, httpOptions, treq, newResponseWriter(w))
+	err = h.Handler.Handle(ctx, httpOptions, treq, newResponseWriter(w))
+
+	// The handler is well behaved and stopped work on context deadline. We
+	// forward this information to the client diligently.
+	if err != nil && err == context.DeadlineExceeded && err == ctx.Err() {
```

Comment:
https://golang.org/ref/spec#Comparison_operators

> Interface values are comparable. Two interface values are equal if they have identical dynamic types and equal dynamic values or if both have value nil.

So != nil kinda shorcuts the costly interface comparaison?

Ok ok, Probably not costly, err != nil removed.